sky/adaptors/aws.py

```diff
@@ -98,21 +98,31 @@ def resource(service_name: str, **kwargs):
 
     Args:
         service_name: AWS resource name (e.g., 's3').
-        kwargs: Other options.
+        kwargs: Other options. We add max_attempts to the kwargs instead of
+            using botocore.config.Config() because the latter is not hashable.
     """
```

Comment:
How about we assert kwargs values by `isinstance(.., typing.Hashable)`? To prevent such an error from occurring. This way L101-102 doesn't need to single out "max_attempts".

Reply:
Sorry for the confusion, it is hashiable, but every object will have a different hash even if the config is the same